### ISSUE TITLE
Null warnings not reported for iterator methods

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#23701]
`yield return null;` should generate `warning: cannot convert null to non-nullable reference`:
```
using System.Collections.Generic;
class C
{
    static IEnumerable<object> F()
    {
        yield return null;
    }
    static IEnumerable<T> G<T>() where T : class
    {
        yield return null;
    }
}
```
